packages/mobile/src/qrcode/utils.ts

```diff
@@ -95,7 +97,20 @@ export function* handleBarcode(
         displayId: data.e164PhoneNumber,
       }
 
-  yield put(storeLatestInRecents(recipient))
+  // TODO data in the QR should be encrypted so noone can create an arbitrary QR with any phone and address combination
+  // add phone number to the recipientCache
+  recipientCache[data.e164PhoneNumber] = {
+    ...recipient,
+    kind: RecipientKind.Contact,
+    contactId: recipient.contactId ? recipient.contactId : uuidv1(),
```

Comment:
This field is represents the contact id in the phone's contact list. We don't use it as an index so we shouldn't populate it with a random uid if it's empty

Reply:
I should leave an empty string then?